runtime/jnichk/jnicheck.c

```diff
@@ -548,21 +548,25 @@ jniCheckCallA(const char* function, JNIEnv* env, jobject receiver, UDATA methodT
 void
 jniCheckArgs(const char *function, int exceptionSafe, int criticalSafe, J9JniCheckLocalRefState *refTracking, const U_32 *descriptor, JNIEnv *env, ...)
 {
-	J9JavaVM *vm = ((J9VMThread*)env)->javaVM;
-	PORT_ACCESS_FROM_JAVAVM(vm);
+	PORT_ACCESS_FROM_JAVAVM(globalJavaVM);
 	va_list va;
 	const U_32 *code;
 	int argNum = 2;
-	UDATA trace = vm->checkJNIData.options & JNICHK_TRACE;
-	UDATA warn = (0 == (vm->checkJNIData.options & JNICHK_NOWARN));
-	J9VMThread * vmThread = (J9VMThread*)env;
+	UDATA trace = globalJavaVM->checkJNIData.options & JNICHK_TRACE;
+	UDATA warn = (0 == (globalJavaVM->checkJNIData.options & JNICHK_NOWARN));
+	J9VMThread *vmThread = (J9VMThread*)env;
+	J9VMThread *currentVMThread = globalJavaVM->internalVMFunctions->currentVMThread(globalJavaVM);
 
-	if (vm->reserved1_identifier != (void*)J9VM_IDENTIFIER) {
-		jniCheckFatalErrorNLS(env, J9NLS_JNICHK_INVALID_ENV, function);
+	if (NULL == currentVMThread) {
+		jniCheckFatalErrorNLS((JNIEnv*)currentVMThread, J9NLS_JNICHK_THREAD_NOT_ATTACHED_TO_VM, function);
+	}
+
+	if (currentVMThread != vmThread) {
+		jniCheckFatalErrorNLS((JNIEnv*)currentVMThread, J9NLS_JNICHK_WRONG_ENV, function);
 	}
 
-	if (vm->internalVMFunctions->currentVMThread(vm) != vmThread) {
-		jniCheckFatalErrorNLS(env, J9NLS_JNICHK_WRONG_ENV, function);
+	if (((J9VMThread*)env)->javaVM != globalJavaVM) {
```

Comment:
Actually, since we've verified that the thread is not NULL and is the current thread, this check is entirely redundant.